controllers/argocd/route.go

```diff
@@ -100,60 +100,19 @@ func (r *ReconcileArgoCD) reconcileGrafanaRoute(cr *argoproj.ArgoCD) error {
 	if argoutil.IsObjectFound(r.Client, cr.Namespace, route.Name, route) {
 		if !cr.Spec.Grafana.Enabled || !cr.Spec.Grafana.Route.Enabled {
 			// Route exists but enabled flag has been set to false, delete the Route
-			return r.Client.Delete(context.TODO(), route)
+			return nil
 		}
```

Comment:
Same as above, I think this `Delete` call should still be here, so that it deletes an old Grafana Route that still exists, even though we have deprecated the functionality.